bootstrap/kubeadm/internal/ignition/clc/clc.go

```diff
@@ -0,0 +1,357 @@
+      enabled: true
+      contents: |
+        [Unit]
+        Description = Mount {{ $label }}
+
+        [Mount]
+        What={{ $disk }}
+        Where={{ $mountpoint }}
+        Options={{ Join $mountOptions "," }}
+
+        [Install]
+        WantedBy=multi-user.target
+    {{- end }}
+storage:
+  {{- if .DiskSetup }}{{- if .DiskSetup.Partitions }}
+  disks:
+    {{- range .DiskSetup.Partitions }}
+    - device: {{ .Device }}
+      {{- if .Overwrite }}
+      wipe_table: {{ .Overwrite }}
+      {{- end }}
+      {{- if .Layout }}
+      partitions:
+      - {}
+      {{- end }}
+    {{- end }}
+  {{- end }}{{- end }}
+  {{- if .DiskSetup }}{{- if .DiskSetup.Filesystems }}
+  filesystems:
+    {{- range .DiskSetup.Filesystems }}
+    - name: {{ .Label }}
+      mount:
+        device: {{ .Device }}
+        format: {{ .Filesystem }}
+        label: {{ .Label }}
+        {{- if .ExtraOpts }}
+        options:
+          {{- range .ExtraOpts }}
+          - {{ . }}
+          {{- end }}
+        {{- end }}
+    {{- end }}
+  {{- end }}{{- end }}
+  files:
+    {{- range .Users }}
+    {{- if .Sudo }}
+    - path: /etc/sudoers.d/{{ .Name }}
+      mode: 0600
+      contents:
+        inline: |
+          {{ .Name }} {{ .Sudo }}
+    {{- end }}
+    {{- end }}
+    {{- if .UsersWithPasswordAuth }}
+    - path: /etc/ssh/sshd_config
+      mode: 0600
+      contents:
+        inline: |
+          # Use most defaults for sshd configuration.
+          Subsystem sftp internal-sftp
+          ClientAliveInterval 180
+          UseDNS no
+          UsePAM yes
+          PrintLastLog no # handled by PAM
+          PrintMotd no # handled by PAM
+
+          Match User {{ .UsersWithPasswordAuth }}
+            PasswordAuthentication yes
+    {{- end }}
+    {{- range .WriteFiles }}
+    - path: {{ .Path }}
+      # Owner
+      #
+      # If Encoding == gzip+base64 || Encoding == gzip
+      # compression: true
+      #
+      # If Encoding == gzip+base64 || Encoding == "base64"
+      # Put "!!binary" notation before the content to let YAML decoder treat data as
+      # base64 data.
+      #
+      {{ if ne .Permissions "" -}}
+      mode: {{ .Permissions }}
+      {{ end -}}
+      contents:
+        inline: |
+          {{ .Content | Indent 10 }}
+    {{- end }}
+    - path: /etc/kubeadm.sh
+      mode: 0700
+      contents:
+        inline: |
+          #!/bin/bash
+          set -e
+          {{ range .PreKubeadmCommands }}
+          {{ . }}
+          {{- end }}
+
+          {{ .KubeadmCommand }}
+          mkdir -p /run/cluster-api && echo success > /run/cluster-api/bootstrap-success.complete
+          mv /etc/kubeadm.yml /tmp/
+          {{range .PostKubeadmCommands }}
+          {{ . }}
+          {{- end }}
+    - path: /etc/kubeadm.yml
+      mode: 0600
+      contents:
+        inline: |
+          ---
+          {{ .KubeadmConfig | Indent 10 }}
+    {{- if .NTP }}{{- if and .NTP.Enabled .NTP.Servers }}
```

Comment:
Adding a  documentation in terms of what files will be genereated and corresponding systemd unit (`kubeadm.service`) maybe helpful for people to understand what to expect and be able customize the behavoir if needed (e.g disable the default bootstrap logic and using customized one )

something like:

- kubeadm.service will be excute `/etc/kubeadm.sh` if `/etc/kubeadm.yml` exsits. 
- /etc/kubeadm.sh
- /etc/kubeadm.yml

To override the behavoir, user can override the kubeadm.service will a drop-in.

Reply:
Where such documentation should be placed? I'm happy to write it down.

Reply:
I'm not certain where is the best place to put this doc and I noticed that you also mentioned the documentation in the PR comments. For this specific doc, I hope it is not too bad an idea just putting it in the code - either in this file for in the package level. If there would be a doc covering the overall ingition support, putting it there makes sense to me too.

Reply:
Done. I added the documentation at package level.

Reply:
Will mark as resolved.